### ISSUE TITLE
Order of search results is not predictable

### ISSUE THREAD

**Issue by kumar303** [BurntSushi/ripgrep#636]
If I re-run the same search on the same codebase, sometimes the results are returned in a different order. 

For example, this makes it hard to do a refactoring where I'm renaming a function and I need to re-run `rg` after fixing each file -- i.e. I can lose my place in the sequence.

```
$ rg --version
ripgrep 0.6.0
-AVX -SIMD
```

**Comment by BurntSushi** [comment]
Duplicate of #152 

Summary: Technically, this bug exists in all recursive directory searchers, but if traversal is single threaded, then in practice, you get deterministic results (but it's not guaranteed). However, ripgrep parallelizes traversal itself, which makes the output far less deterministic than it would be otherwise.

TL;DR - Pick one: speed or determinism. Speed is the default. If you want determinism, then use the `--sort-files` flag.
